### PR TITLE
API for signing metadata on write

### PR DESCRIPTION
**Pull request by kimcs**

### CODE REVIEW

File: src/main/proto/data-access-api.proto [hunk 1]
```diff
@@ -4,73 +4,60 @@ option java_multiple_files = true;
...
-    // Find version that was valid of the time of snapshot.
-    // If not supplied (or value == 0) will find timestamp of latest version.
-    int64 snapshot = 4;
-
-    // If Privilege is a WRITE operation, then the writeOptions must be present
-    // and contain the desired valuation and state
-    WriteOptions writeOptions = 5;
+message WriteLocationRequest {
+    // A json serialized representation of metadata according to the dataset-api specification
+    string metadataJson = 6;
```
Comment (bjornandre) [inline src/main/proto/data-access-api.proto]: Should it be 
`string metadataJson = 1`

File: src/main/proto/data-access-api.proto [hunk 2]
```diff
@@ -4,73 +4,60 @@ option java_multiple_files = true;
...
 }
 
-message LocationResponse {
-    string parentUri = 1; // only set if accessAllowed is true, otherwise an empty string
-    string version = 2; // only set if accessAllowed is true and there was a READ request, otherwise an empty string
+message WriteLocationResponse {
     bool accessAllowed = 3;
```
Comment (bjornandre) [inline src/main/proto/data-access-api.proto]: I suggest reindexing the params 1-3

Comment (hadrienk) [reply src/main/proto/data-access-api.proto]: You should mark the field as reserved.
https://developers.google.com/protocol-buffers/docs/proto#reserved

File: src/main/proto/data-access-api.proto (hunk 1 above)
Comment (kimcs) [reply src/main/proto/data-access-api.proto]: har ikke egentlig noe å si ;)

File: src/main/proto/data-access-api.proto (hunk 2 above)
Comment (kimcs) [reply src/main/proto/data-access-api.proto]: ok :p

**Pull request merged**